lib/CMakeLists.txt

```diff
@@ -41,6 +41,28 @@ add_library(libmosquitto SHARED
 	util_mosq.c util_mosq.h
 	will_mosq.c will_mosq.h)
 
+#target for building static version of library
```

Comment:
I agree, that would reduce the duplication here. I elected to name it `C_SRC` however because here it's just all the C code, and so if one wanted to expand the C++ to use this convention, one could use a similar name such as `CXX_SRC` or something.